### ISSUE TITLE
Example usage with tests.

### ISSUE THREAD

**Issue by Nsttt** [QuiiBz/next-international#36]
Hello!
Been using this library for a while but I'm having a bit of a catch up when trying to make unit tests with it.

On regular i18n I either mock the tests or pass a default locale to use with it. But in this case due to passing props using Next.js `pageProps` I'm running into some issues.

Here are the few pieces of code I'm currently using to implement my tests.

```tsx
import { createI18n } from "next-international";
import Locale from "../../locales/en";

const TestWrapper = ({ children }: { children: JSX.Element }) => {
  const { I18nProvider } = createI18n<typeof Locale>({
    en: () => import("../../locales/en"),
  });
 
  return (
      <I18nProvider locale={Locale}>
         {children}
      </I18nProvider>
  );
};
 
export default TestWrapper;
```
And then I wrap a component with it.

```tsx
import { render, screen, waitFor } from "@testing-library/react";
import TestWraper from "../utils/testWraper";
import HomePage from "../../pages/[building]/homePage";

describe("Homepage tests", () => {
  beforeEach(() => {
    render(
      <TestWraper>
        <HomePage />
      </TestWraper>,
    );
  });
  it("Render logout button", () => {
    expect(
      screen.getByRole("heading", {
        name: /logout/i,
      }),
    ).toBeInTheDocument();
  });
});
```
And I get this error.
```
 console.error
      Error: Uncaught [TypeError: Cannot destructure property 'locale' of '(0 , B.useRouter)(...)' as it is null.]
          at reportException.
```

It might be handy if there's some sort of example on how to write a basic test with it.

**Comment by QuiiBz** [comment]
Hi! Looks like `useRouter` returns null because it's coming from Next.js. You can take a look at our tests here, where we mock `useRouter`:

https://github.com/QuiiBz/next-international/blob/09675c668f5c59a5caf551dbf657b8054d77a1d7/packages/next-international/__tests__/i18n-provider.test.tsx#L8

**Comment by QuiiBz** [comment]
Yeah it might be a good idea to add it in the [example section](https://github.com/QuiiBz/next-international#examples) of the README. Feel free to send a PR if you want :)

**Comment by Nsttt** [comment]
Done. #37 . Hope it's properly explained like that. 😄

**Comment by QuiiBz** [comment]
Closing this issue since we merged #37, thanks again!
